File: solution/1000-1099/1048.Longest String Chain/Solution.go
func longestStrChain(words []string) (ans int) {
	sort.Slice(words, func(i, j int) bool { return len(words[i]) < len(words[j]) })
	d := map[string]int{}
	for _, s := range words {
		x := 1
		for i := 0; i < len(s); i++ {
			t := s[:i] + s[i+1:]
			x = max(x, d[t]+1)
		}
		d[s] = x
		ans = max(ans, x)
	}
	return
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

File: solution/1000-1099/1048.Longest String Chain/Solution.py
class Solution:
    def longestStrChain(self, words: List[str]) -> int:
        words.sort(key=len)
        d = defaultdict(int)
        ans = 0
        for s in words:
            x = 1
            for i in range(len(s)):
                t = s[:i] + s[i + 1:]
                x = max(x, d[t] + 1)
            d[s] = x
            ans = max(ans, x)
        return ans
